Test cleanup

richardschneider opened the pull request:
These are changes I've made to keep both Sprinkler and Blaze happy.

richardschneider committed:
Two BI02 test names.

richardschneider committed:
Wrong Content-Type for a bundle.

richardschneider committed:
Merge branch 'master' of https://github.com/richardschneider/sprinkler

richardschneider committed:
The XDS example document has the wrong ID for the binary resource.

richardschneider committed:
Transactions replace the reference IDs with the resource ID not the resource ID and version number.

richardschneider committed:
Create a questionaire before search for one.

richardschneider committed:
Fix the XSD validation errors.

richardschneider committed:
Updating a resource requires the current version number.

mharthoorn commented:
I've merged all your changes except the bundle ID's. I still have that under review, because our server says that a resource id cannot be the same as a self link.

richardschneider committed:
Skip the Document Mailbox test (MA*) if the server's conformance statement does not support it.

DSTU does not describe the outcome of a POST .../Mailbox.  But both of our server's retrun a Bundle with a reference to a Binary and DocumentReference resource.

The DocumentReference.Location can be relative or absolute URI.

richardschneider committed:
Some servers (blaze) require a version number to be an int.

richardschneider committed:
NoContent is the expected response from an AffixTags operation.

richardschneider committed:
Getting sln to compile

richardschneider committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	Sprinkler/Program.cs
	Sprinkler/Tests/SearchTest.cs

richardschneider commented:
more changes
